[PORT]  Ashwalker breathing and oxygen toxcicity

Flakkyboo opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Ported from https://github.com/Citadel-Station-13/Citadel-Station-13/pull/8159

Currently Ashwalkers do not breathe, they do not care about atmos in any fashion this change aims to fix that by giving them lungs. Ashwalker lungs are only suited to the low pressure environment of lavaland and will begin to suffocate if they go into areas with atmosphere suited for humans.

This also makes it so if you are in an environment with too high an oxygen concentration you will begin to get dizzy and take suffocation damage gradually until you leave get some better air.

It is now compatible with the randomized lavaland atmosphere. (many thanks to St0rm for the assist)

## Why It's Good For The Game
This is step one in a process to make ashwalkers more than a card to murderbone and more of a living species on a planet.

While you can still do the unga bunga limitations in atmosphere make it so going up to the station for long periods of time is more of a danger. 

Makes high oxygen environments more dangerous for everyone.

## Changelog
:cl:
add: New Ashwalker lungs "Ash Lungs"
add: Ashwalkers can only breath lavaland atmosphere.
add: Code to allow for suffocation if breathing in too much oxygen.
add: Compatibility with randomized lavaland atmosphere.
add: New sprite for Ashwalker lungs.
add: Code that allows Ashwalkers to process BZ moles very quickly in the form of a minuscule heal
tweak: Removed unneeded code related to plasmamen suffocating from too much oxygen as they already suffocate without plasma.
tweak: Added a maximum oxygen threshold for cybernetic lungs.
tweak: Adjusted the randomized lavaland atmos base values to prevent situations where o2 or n2
goes above ashwalker maximums.
Fix: Lava tiles around necropolis tendril now use lavaland atmos 
Fix: Fixed conflicts with beepeople lungs
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Flakkyboo committed:
Ashwalker breathing and oxygen toxcitiy

Gives ashwalkers some shiny new lungs suited to their environment

Flakkyboo committed:
Ashwalker breathing and oxygen toxcitiy

Gives ashwalkers some shiny new lungs suited to their environment

Flakkyboo committed:
Ashwalker breathing and oxygen toxicity 

New Updated surgery.dmi

Flakkyboo committed:
Ashwalker breathing and oxygen toxicity 

New Updated surgery.dmi

Flakkyboo committed:
Ashwalker breathing and oxygen toxicity 

new surgery.dmi file added. old file overwritten

Flakkyboo committed:
Ashwalker breathing and oxygen toxicity 

new surgery.dmi file added. old file overwritten

Flakkyboo committed:
Merge remote-tracking branch 'Flakkyboo/ashlizard-changes' into ashlizard-changes

Flakkyboo committed:
Ashwalker changes and oxygen toxicity 

re-adding removed surgery.dmi file

Flakkyboo committed:
Ashlizard breathing and oxygen toxicity

re-added and commented out code relating to randomized lavaland atmos as requested by crossed

Flakkyboo committed:
Merge remote-tracking branch 'Flakkyboo/ashlizard-changes' into ashlizard-changes

St0rmC4st3r commented:
Don't revert the random lavaland atmos code. Just make them either immune to secondary gasses or make something like BZ beneficial to them.

ike709 reviewed (changes requested):
> del: reverted code related to randomized lavaland atmosphere

This PR isn't getting merged as long as random atmos is removed. Have ashwalkers breathe whatever the gasses of the round are.

Flakkyboo committed:
Ashlizard breathing and oxygen toxicity 

reintroduced randomized lavaland code in preparation for making it compatible with ashwalker lungs.

St0rmC4st3r committed:
Should make BZ benefitial

Flakkyboo committed:
Merge pull request #1 from St0rmC4st3r/ashlizard-changes

Should make BZ beneficial

Flakkyboo committed:
ashlizard breathing and oxygen toxicity 

Final changes and some bugfixes with the BZ change

Flakkyboo committed:
Ashlizard breathing and oxygen toxcicty

Fixed ashwalker nest atmos being lethal to ashwalker.
adjusted random atmos n2 values

ike709 commented on code/game/turfs/open.dm:
```diff
@@ -87,7 +87,9 @@
 
 /turf/open/indestructible/necropolis/air
-	initial_gas_mix = OPENTURF_DEFAULT_ATMOS
+	//initial_gas_mix = OPENTURF_DEFAULT_ATMOS
```
Don't comment out code, just remove it.

ike709 commented on code/game/turfs/open.dm:
```diff
@@ -97,7 +99,8 @@
 
 /turf/open/indestructible/boss/air
-	initial_gas_mix = OPENTURF_DEFAULT_ATMOS
+	//initial_gas_mix = OPENTURF_DEFAULT_ATMOS
```
Don't comment out code, just remove it.

ike709 commented on code/game/turfs/simulated/lava.dm:
```diff
@@ -168,6 +168,8 @@
 	canSmoothWith = list(/turf/open/lava/smooth)
 
+
+
```
Random whitespace addition?

Flakkyboo replied:
ah that was what was left over from my hacky fix of the nest atmos before i used fastdmm.
forgot to get rid of the extra lines after i got rid of the commented out initial atmos.

thanks for looking over this ill make those changes now

ike709 commented on code/modules/mob/living/carbon/life.dm:
```diff
@@ -167,8 +168,21 @@
+		SEND_SIGNAL(src, COMSIG_ADD_MOOD_EVENT, "suffocation", /datum/mood_event/suffocation)
+
+
 	if(O2_partialpressure < safe_oxy_min) //Not enough oxygen
```
```suggestion
	else if(O2_partialpressure < safe_oxy_min) //Not enough oxygen
```

Flakkyboo replied:
applied these changes

ike709 commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -106,11 +107,25 @@
 	//Too much oxygen! //Yes, some species may not like it.
 	if(safe_oxygen_max)
-		if(O2_pp > safe_oxygen_max)
+		if((O2_pp > safe_oxygen_max) && safe_oxygen_max ==0)
```
This check will never pass.

Flakkyboo replied:
I ended up removing these checks "&& safe_oxygen_max ==0)" as every race that wasnt plasmamen it worked with and plasmamen already had a hard time with oxygen and didnt need to take more suffocation damage

ike709 commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -106,11 +107,25 @@
 			H.apply_damage_type(CLAMP(ratio, oxy_breath_dam_min, oxy_breath_dam_max), oxy_damage_type)
 			H.throw_alert("too_much_oxy", /obj/screen/alert/too_much_oxy)
+
+		else if((O2_pp > safe_oxygen_max) && !(safe_oxygen_max == 0))
```
What

ike709 commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -444,8 +464,27 @@
 	cold_level_2_threshold = 140
 	cold_level_3_threshold = 100
+
+obj/item/organ/lungs/ashwalker
```
```suggestion
/obj/item/organ/lungs/ashwalker
```

Flakkyboo committed:
Ashlizard changes 

committed changes recommended by review and removed unneeded code

Flakkyboo committed:
ashwalker  breathing and oxygen toxcicty

This should fix the conflict with the beepeople.
Adjusted the ashwalker temperature thresholds again

Flakkyboo committed:
Merge branch 'master' into ashlizard-changes

Flakkyboo committed:
Ashwalker breathing and oxygen toxcicty

Fixing a typo in the changes make to Apid lungs when fixing the previous conflict

Flakkyboo committed:
Merge remote-tracking branch 'Flakkyboo/ashlizard-changes' into ashlizard-changes

ike709 commented:
After careful consideration and previous attempts to make this code functional, we've determined that this PR will not be acceptable without considerable effort from maintainers.

We recommend smaller PRs in the future until you have a better understanding of coding.
